corehq/apps/style/static/style/ko/knockout_bindings.ko.js

```diff
@@ -629,6 +629,53 @@ ko.bindingHandlers.typeahead = {
     }
 };
 
+/**
+ * Converts the bound element to a select2 widget. The value of the binding is
+ * a list of strings, or a list of objects with the keys 'id' and 'text' used
+ * for the select2's options.
+ */
+ko.bindingHandlers.select2 = new function(){
```

Comment:
I wanted to create an object where both the `init` and `update` properties refer to the `SOURCE_KEY` property, so I couldn't use an object literal. I think the anonymous constructor approach that I used looks a little cleaner than either breaking the object's definition over multiple statements or using a "self-calling" anonymous function. Is there a best practice or style guideline for this situation?
(I found a somewhat related discussion on [SO](http://stackoverflow.com/questions/1536127/module-pattern-vs-instance-of-an-anonymous-constructor))